run.sh

```diff
@@ -1,67 +1,110 @@
+        -h | Outputs this help text"
+}
+
+if [ "${1}" == "-h" ]; then
+    usage
+    exit 0
+fi
+
+if [ "$#" -ne 0 ]; then
+    usage
+    exit 1
+fi
 
 # see init-softhsm.sh for slot initialization
+export SOFTHSM2_CONF="${PWD}/softhsm2.conf"
+echo "directories.tokendir = ${PWD}/softhsm/" > ${SOFTHSM2_CONF}
 
-# Simulating intermediate HSM
-ceremony --config e1-key.yaml
-ceremony --config e2-key.yaml
-ceremony --config r3-key.yaml
-ceremony --config r4-key.yaml
-ceremony --config r3-cross-csr.yaml
-ceremony --config r4-cross-csr.yaml
-
-# Verify the self-signature on these CSRs.
-openssl req -verify -in int-r3.cross-csr.pem -noout
-openssl req -verify -in int-r4.cross-csr.pem -noout
-
-# Simulating root HSM
-ceremony --config root-x1.yaml
-ceremony --config root-x2.yaml
-ceremony --config e1-cert.yaml
-ceremony --config e2-cert.yaml
-ceremony --config r3-cert.yaml
-ceremony --config r4-cert.yaml
-ceremony --config root-x1.crl.yaml
-ceremony --config root-x2.crl.yaml
-ceremony --config x2-signed-by-x1.yaml
-
-# Verify the root -> intermediate signatures, plus the TLS Server Auth EKU.
-# -check_ss_sig means to verify the root certificate's self-signature.
-# 1609459200 is January 1 2021; this is necessary because we're testing with NotBefore in the future.
-openssl verify -check_ss_sig -attime 1609459200 -CAfile root-x2.cert.pem -purpose sslserver int-e1.cert.pem int-e2.cert.pem
-openssl verify -check_ss_sig -attime 1609459200 -CAfile root-x1.cert.pem -purpose sslserver int-r3.cert.pem int-r3.cert.pem
-
-# Verify the X1 -> X2 cross-signature.
-# Don't verify `-purpose sslserver` here because x2-signed-by-x1 intentionally
-# doesn't have the "TLS Server Auth" EKU (and doesn't need it).
-openssl verify -check_ss_sig -attime 1609459200 -CAfile root-x1.cert.pem x2-signed-by-x1.cert.pem
-
-# Verify the path from X1 -> X2 -> E1 and X1 -> X2 -> E2, plus the TLS Server Auth EKU.
-openssl verify -check_ss_sig -attime 1609459200 -CAfile root-x1.cert.pem -purpose sslserver -untrusted x2-signed-by-x1.cert.pem int-e1.cert.pem
-openssl verify -check_ss_sig -attime 1609459200 -CAfile root-x1.cert.pem -purpose sslserver -untrusted x2-signed-by-x1.cert.pem int-e2.cert.pem
-
-# Verify the CRLs.
-openssl crl -verify -CAfile root-x1.cert.pem -in root-x1.crl.pem -noout
-openssl crl -verify -CAfile root-x2.cert.pem -in root-x2.crl.pem -noout
-
-rm root-x1.key.pem
-rm root-x1.cert.pem
-
-for c in *.cert.pem ; do
-  openssl x509 -text -noout -out $c.txt -in $c
-done
-for c in *.crl.pem ; do
-  openssl crl -inform pem -in $c  -text -noout > $c.txt
+# Store the output in a ramdisk so we don't chew up my disk endlessly running this tooling.
+RAMDISK_DIR="/run/shm/ceremonies"
+mkdir -p "${RAMDISK_DIR}"
+for ceremonyYear in $(find ./ceremonies/ -maxdepth 1 -type d -printf '%P '); do
+    mkdir -p "${RAMDISK_DIR}/${ceremonyYear}"
 done
+if [ ! -L "ceremony-output" ]; then
+    ln -s "${RAMDISK_DIR}/" ceremony-output
+fi
+
+function setup_ceremony_tools() {
```

Comment:
What if -- and feel free to say "nah that's too much" -- each subdirectory's run.sh knew the correct version of the ceremony tool to fetch, and handled that itself? We could use the version that was actually used in the historical ceremonies (except for 2000 and 2015, for which it didn't exist yet).

Maybe that's just worth filing a future bug about.

Reply:
Aside from [my recent changes to ceremony](https://github.com/letsencrypt/boulder/pull/7005), it hadn't really been touched in a meaningful way that would impact previous ceremonies hence just using the one old version. I get what you're saying though and I'm coming around to it.